### PR TITLE
Add new public actions, move existing actions to internal actionCmd

### LINKED ISSUE #59: Add disable property to Actions
Follow up of https://github.com/gluonhq/rich-text-area/pull/53/files#r812390728

### PR DESCRIPTION
**Pull request by jperedadnr**
Fixes #59

The new Actions are non static, and get the viewmodel when control is added to scenegraph. Existing actions are renamed to ActionCmd, and treated as internal.

### CODE REVIEW

**Comment by jperedadnr** [comment]
@eugener No pressure, but any chance you can have a look at this PR? 
I have ongoing changes for other issues that depend on it and it'd be better not to pile them in the same PR.

**Comment by tiainen** [comment]
Instead of making an `AbstractAction` class and adding an implementation for each `ActionCmd`, we could simplify it by making `AbstractAction` concrete (renaming it to `BasicAction`?) and directly passing an instance of `Function<Action, ActionCmd>` to the constructor.

In `ActionFactory`, you would for instance create the `UndoAction` as follows:

    undo = new BasicAction(control, action -> ACTION_CMD_FACTORY.undo());

The `BasicAction` constructor looks like this:

    public BasicAction(RichTextArea control, Function<Action, ActionCmd> fnGetActionCmd) {
        this.fnGetActionCmd = fnGetActionCmd;
        ...
    }

And then the `BasicAction.getActionCmd()` method can become this:

    private ActionCmd getActionCmd() {
        fnGetActionCmd.apply(this);
    }

This would also remove the need for the enum `Action.ActionType` and the `BasicAction.ACTION_MAP` variable.

File: src/main/java/com/gluonhq/richtext/viewmodel/ActionCmd.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.gluonhq.richtext.viewmodel;
+
+import javafx.beans.binding.BooleanBinding;
+
+public interface ActionCmd {
+    void apply(RichTextAreaViewModel viewModel);
```
Comment (tiainen) [inline src/main/java/com/gluonhq/richtext/viewmodel/ActionCmd.java:6]: Would `execute` be a better name for a command?

Comment (jperedadnr) [reply src/main/java/com/gluonhq/richtext/viewmodel/ActionCmd.java:6]: Yes, changed

**Comment by jperedadnr** [comment]
@tiainen that's a nice refactor indeed, it really simplifies it.

File: src/main/java/com/gluonhq/richtext/action/ActionFactory.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    private final RichTextArea control;
+
+    public ActionFactory(RichTextArea control) {
+        this.control = control;
+    }
+
+    private Action undo;
+    public Action undo() {
+        if (undo == null) {
+            undo = new UndoAction(control);
```
Comment (tiainen) [inline src/main/java/com/gluonhq/richtext/action/ActionFactory.java:19]: All these inner classes can be replaced now by directly constructing a `BasicAction`:

```suggestion
            undo = new BasicAction(control, action -> ACTION_CMD_FACTORY.undo());
```

Comment (jperedadnr) [reply src/main/java/com/gluonhq/richtext/action/ActionFactory.java]: done

**Review by tiainen — APPROVED** [review]

**Pull request merged**